### PR TITLE
Update App Launcher Create App Form based on new design

### PR DESCRIPTION
**Pull request by jbouder**
## Reference Issues or PRs

N/A

## What does this implement/fix?

_Put a `x` in the boxes that apply_

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds a feature)
- [ ] Breaking change (fix or feature that would cause existing features not to work as expected)
- [ ] Documentation Update
- [ ] Code style update (formatting, renaming)
- [x] Refactoring (no functional changes, no API changes)
- [ ] Build related changes
- [ ] Other (please describe):

## Testing

- [x] Did you test the pull request locally?
- [x] Did you add new tests?

## Screenshots
Screenshot 2024-03-05 at 7 49 36 PM
Screenshot 2024-03-05 at 7 49 48 PM
Screenshot 2024-03-05 at 7 49 57 PM
Screenshot 2024-03-05 at 8 45 59 PM

## Any other comments?
- Moved app create and edit form to its own page
- Updated App Form to utilize MUI components
- Updates to Server Types page for new app forms
- Misc styling updates
- Fixed issue causing env variables to load incorrectly into edit form

### CODE REVIEW

File: ui/src/App.tsx [hunk 1]
```diff
@@ -2,19 +2,15 @@ import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
...
 import { JhData } from './types/jupyterhub';
 import { getJhData } from './utils/jupyterhub';
 
-const queryClient = new QueryClient({
-  defaultOptions: {
-    queries: {
-      refetchOnWindowFocus: false,
-    },
-  },
-});
```
Comment (jbouder) [inline ui/src/App.tsx:17]: This is no longer needed given the move to separate pages

File: jhub_apps/service/japps_routes.py [hunk 2]
```diff
@@ -0,0 +1,14 @@
+from fastapi import APIRouter, FastAPI, Request
+from fastapi.responses import HTMLResponse
+from fastapi.templating import Jinja2Templates
+
+app = FastAPI()
+templates = Jinja2Templates(directory="jhub_apps/templates")
```
Comment (aktech) [inline jhub_apps/service/japps_routes.py:6]: This path would not be accessible on the k8s environment, use this rather: https://github.com/nebari-dev/jhub-apps/blob/9d37c2f6e157758c0cd68d71a721521c5759bcc0/jhub_apps/__init__.py#L7

You may have seen that here: https://github.com/nebari-dev/jhub-apps/blob/9d37c2f6e157758c0cd68d71a721521c5759bcc0/jhub_apps/service/routes.py#L46

but that's a leftover from a cleanup and not used anywhere.

File: jhub_apps/tests_e2e/test_integration.py [hunk 3]
```diff
@@ -50,12 +50,13 @@ def test_panel_app_creation(playwright: Playwright) -> None:
...
-        page.get_by_label("Display Name *").fill(app_name)
+        page.get_by_label("Name *").click()
+        page.get_by_label("Name *").fill(app_name)
         logger.info("Select Framework")
-        page.get_by_label("Framework *").select_option(framework)
+        page.locator("id=framework").click()
+        page.get_by_role("option", name="Panel").click()
         logger.info("Click Submit")
-        page.get_by_role("button", name="Submit").click()
+        page.get_by_role("button", name="Create App").click()
```
Comment (aktech) [inline jhub_apps/tests_e2e/test_integration.py:59]: Awesome! 🎉

File: jhub_apps/service/japps_routes.py (hunk 2 above)
Comment (jbouder) [reply jhub_apps/service/japps_routes.py]: Fixed

**Review by aktech — CHANGES_REQUESTED** [review]
I have a minor comment, other than that it looks great to me! 🎉

File: jhub_apps/service/japps_routes.py [hunk 4]
```diff
@@ -0,0 +1,19 @@
+from fastapi import APIRouter, FastAPI, Request
+from fastapi.responses import HTMLResponse
+from fastapi.templating import Jinja2Templates
+
+
+from jhub_apps import HERE
```
[the remark this answers is not in the document]
Comment (jbouder) [reply jhub_apps/service/japps_routes.py]: Updated as requested

**Review by aktech — APPROVED** [review]

**Pull request merged**